### PR TITLE
Patch Perl for macOS 11.x

### PR DESCRIPTION
**Pull request by BytesGuy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Perl will currently fail to build (Intel and Apple Silicon) as it is expecting to match the product version `10.x`, whereas Big Sur has been bumped to `11.0`. This is a temporary patch until support is available upstream (PR created: https://github.com/Perl/perl5/pull/17946)

### CODE REVIEW

**Comment by Bo98** [comment]
Makes sense. Let's give upstream a couple days to give comment first (PR diff URLs aren't stable as they change if any changes to the PR is made).

**Comment by BytesGuy** [comment]
> Makes sense. Let's give upstream a couple days to give comment first (PR diff URLs aren't stable as they change if any changes to the PR is made).

Ok, good call 🙂 Will follow up on this when my PR is reviewed! Meanwhile this might help anyone who is blocked. 

I wonder how many of these kinds of issues we will see since being bumped to `11.0`!

**Comment by coyotanark** [comment]
https://eclecticlight.co/2020/07/21/big-sur-is-both-10-16-and-11-0-its-official/

**Comment by Bo98** [comment]
We won’t be enabling `SYSTEM_VERSION_COMPAT` in Homebrew, nor do we allow using an older SDK.

`SYSTEM_VERSION_COMPAT` could be applied to certain troublesome formulae where a patch to support 11.x is non-trivial, but this is a short term workaround and not a long term fix. Early indications show that the 11.x problem is not really as common as feared.

**Comment by BytesGuy** [comment]
I think this is the only instance of an issue with the 11.0 version number so far. I’ll be following up on the PR to the Perl repo later, been offline for a few days so just catching up with things.

File: Formula/perl.rb
Comment (jonchang) [inline Formula/perl.rb:23]: Please either use a commit URL or submit this to formula-patches. Note that the `sha256` hash is already mismatched as you've pushed additional commits to that branch.

**Review by jonchang — CHANGES_REQUESTED** [review]

File: Formula/perl.rb
Comment (BytesGuy) [reply Formula/perl.rb:23]: @jonchang Unless it's urgent we can wait until the upstream PR is merged in and use that as the patch, otherwise it just means updating the formula again. I'm pushing changes to the upstream PR shortly so hopefully it will be good enough to get merged in to resolve the issue for now.

Comment (BytesGuy) [reply Formula/perl.rb:23]: Just pushed changes so let's hold for a few days to see what happens. If this doesn't get merged soon I'll just create a formula-patch as it will be easier to manage 👍

**Comment by BytesGuy** [comment]
@claui @jonchang I’ve opened https://github.com/Homebrew/formula-patches/pull/293 so we can resolve this without waiting for the PR to be merged into Perl, it is taking some time!

**Comment by claui** [comment]
> I’ve opened https://github.com/Homebrew/formula-patches/pull/293 so we can resolve this without waiting for the PR to be merged into Perl, it is taking some time!

Thanks! Merged the formula patch.

**Comment by BytesGuy** [comment]
> > I’ve opened [Homebrew/formula-patches#293](https://github.com/Homebrew/formula-patches/pull/293) so we can resolve this without waiting for the PR to be merged into Perl, it is taking some time!
> 
> Thanks! Merged the formula patch.

Thanks 🙂 Just updated the formula to point to that patch, should be all good now!

File: Formula/perl.rb
Comment (claui) [inline Formula/perl.rb:21]: I prefer to always start that comment with the literal word _Remove_ so it’s 1. unambiguously actionable, and 2. on the off chance it’s forgotten, it can be grepped.

What do you think?

Comment (BytesGuy) [reply Formula/perl.rb:21]: @claui Sounds good to me, feel free to change it 👍

Comment (claui) [reply Formula/perl.rb]: @BytesGuy Done.
Fyi, I also changed the Git revision of the formula patch to a plain commit revision.

Comment (BytesGuy) [reply Formula/perl.rb]: @claui Thank you 🎉

**Comment by Renaitare** [comment]
I just wanted to ask.
Any update on the merge status of this PR yet? It seems complete and has been for almost a week now.

**Pull request merged**
